internal/store/aof.go

```diff
@@ -25,6 +25,10 @@ const (
 )
 
 func NewAOF(path string) (*AOF, error) {
+	if !config.DiceConfig.Server.PersistenceEnabled {
```

Comment:
Currently, the default behavior is to create the AOF as part of cleanup() or BGREWRITEAOF. Introducing this change could result in silently failing to persist without any warning. To avoid introducing new behaviors in this PR, could we please keep the changes within the scope of the config file and implement only the currently used configurations? We can consider addressing this as part of a separate PR.

Reply:
Your suggested changes makes sense to me and will be implemented soon, thanks for reviewing

Reply:
There's new config introduced for this as `writeAOFEnabled` shall we reuse that?